src/esp/sensor/VisualSensor.cpp

```diff
@@ -136,5 +137,58 @@ bool VisualSensor::getObservation(sim::Simulator& sim, Observation& obs) {
   return true;
 }
 
+VisualSensor::MoveSemanticSensorNodeHelper::MoveSemanticSensorNodeHelper(
+    VisualSensor& visualSensor,
+    sim::Simulator& sim)
+    : visualSensor_(visualSensor), sim_(sim) {
+  CORRADE_INTERNAL_ASSERT(visualSensor_.specification()->sensorType ==
+                          SensorType::Semantic);
+  scene::SceneNode& node = visualSensor_.node();
+  CORRADE_ASSERT(
+      !scene::SceneGraph::isRootNode(node),
+      "VisualSensor::moveSemanticSensorToSemanticSceneGraph(): the semantic "
+      "sensor is attached to the root node, and thus cannot be moved.", );
+
+  // check if the sensor is already in this semantic scene graph
+  if (node.scene() == sim.getActiveSemanticSceneGraph().getRootNode().scene()) {
```

Comment:
I think this case will always cause a crash?  Since the destructor will always hit `CORRADE_INTERNAL_ASSERT(semanticSensorParentNodeBackup_);`

Reply:
Well, the basic assumption is that at the very beginning, the semantic sensor is attached to the regular scene graph, not the semantic scene graph. Your comment makes me think if it is always true.

If this is true, then you do not need to worry about it here.

Reply:
If there are NO two scene graphs, e.g., only the semantic scene graph, then user should never ever use `MoveSemanticSensorNodeHelper` at all.

Reply:
Should this be an assert then?  Right now it looks like a feature where this won't move the sensor if it doesn't need to.  However, the destructor assumes that if the sensor isn't part of the active scene graph, then the constructor moved it to the semantic scene graph and will try to move it back.  This will cause a crash however.